ores/api.py

```diff
@@ -81,22 +87,31 @@ def _score(self, context, models, rev_ids):
         logging.debug("Starting up thread pool with {0} workers"
                       .format(self.workers))
         with ThreadPoolExecutor(max_workers=self.workers) as executor:
-            futures = []  # A list of future results
+            future_rev_ids = []  # A list of future results and the revids
 
             # This loop loads all rev_id_batch's into the executor for
             # processing
             for rev_id_batch in chunked(rev_ids, self.batch_size):
                 rev_id_batch = list(rev_id_batch)
                 logging.debug("Starting batch of {0} revids"
                               .format(len(rev_id_batch)))
-                futures.append(executor.submit(self._score_request,
-                                               context, rev_id_batch,
-                                               models))
+                future_rev_ids.append((
+                    executor.submit(self._score_request, context, rev_id_batch,
+                                    models),
+                    rev_id_batch))
```

Comment:
I think we can avoid this by pushing down the responsibility for stuffing error result cells.

Reply:
That's a good point.  I suppose we could have _score_request return an iterator of error docs in the case that the whole call fails.

Reply:
Fixed in most recent push.